sympy/sets/sets.py

```diff
@@ -916,6 +916,11 @@ def _complement(self, other):
             b = Interval(self.end, S.Infinity, not self.right_open, True)
             return Union(a, b)
 
+        if isinstance(other, FiniteSet):
```

Comment:
Sorry. Currently
```
>>> FiniteSet(n) - S.Reals
FiniteSet(n)    # this is wrong
```

So using this block of code, i am checking whether `FiniteSet(n)` contains all the `non-number` elements . If `yes` then `None` is to be returned
which results in this:
```
>>> FiniteSet(n) - S.Reals
FiniteSet(n) - S.Reals
```